### PR TITLE
Add a `Metadata` protobuf module

### PR DESCRIPTION
**Pull request by olix0r**
As we introduce HTTP routes (for inbound policy), we need the control
plane to describe resources references for metrics, and other
diagonstics in the proxy. Currently, the control plane uses a free-form
map of strings, but the proxy only references a specific set of keys,
which is not clearly documented via the gRPC API.

This change replaces the `Authz` type's `labels` map with a `Metadata`
type. Controllers should set both fields for at least one stable
version.

Signed-off-by: Oliver Gould <ver@buoyant.io>

### CODE REVIEW

**Comment by hawkw** [comment]
looks like the CI build is broken due to missing imports...but only for docs? seems weird: https://github.com/linkerd/linkerd2-proxy-api/runs/7222796039?check_suite_focus=true

File: proto/inbound.proto [hunk 1]
```diff
@@ -108,7 +109,15 @@ message Authz {
   //
   // A control plane SHOULD return the same keys in all authorizations. That is,
   // we do NOT want to return arbitrary pod labels in this field.
+  //
+  // `labels` should be considered deprecated. `metadata` is preferred. However,
+  // controllers should continue to set `labels` for compatbility with older
```
Comment (hawkw) [inline proto/inbound.proto:114]: typo:
```suggestion
  // controllers should continue to set `labels` for compatibility with older
```

File: proto/inbound.proto [hunk 2]
```diff
@@ -108,7 +109,15 @@ message Authz {
...
+  // proxies.
   map<string, string> labels = 3;
+
+  // If set describes an Authorization configuration resource/implicit. Replaces
```
Comment (hawkw) [inline proto/inbound.proto:118]:
```suggestion
  // If set, describes an Authorization configuration resource/implicit. Replaces
```

**Review by hawkw — COMMENTED** [review]
some docs nits, and i wasn't super clear on the way the `default` variant for `Metadata` is supposed to work. it would be nice to have a bit more of a description of that in the comments. other than that, lgtm!

File: proto/meta.proto [hunk 3]
```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package io.linkerd.proxy.meta;
+
+option go_package = "github.com/linkerd/linkerd2-proxy-api/go/meta";
+
+// General metadata about a configuration object. Typically references either an
+// implicit default configuration or a Kubernetes resource.
```
[the remark this answers is not in the document]
Comment (adleong) [reply proto/meta.proto:8]: based on the current behavior of the policy-controller, I think this is supposed to describe what the default policy is (e.g. "all-authenticated").  But I agree, it's not clear.

File: proto/meta.proto [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+message Metadata {
+  oneof kind {
+    // A name describing a default/implicit configuration.
+    //
+    // For example, a policy default name like `all-authenticated` describes an
+    // implicit controller-implementedc policy that does not exist as a cluster
```
Comment (hawkw) [inline proto/meta.proto:14]: typo
```suggestion
    // implicit controller-implemented policy that does not exist as a cluster
```

**Review by hawkw — APPROVED** [review]
one last typo, but this seems good now

**Review by adleong — APPROVED** [review]

**Pull request merged**